### ISSUE TITLE
Support base64 file import

### ISSUE THREAD

**Issue by sfcrebecca** [surveyjs/surveyjs#972]
### Are you requesting a feature, reporting a bug or ask a question?
Feature

### What is the current behavior?
Our system provides a function for users to download the survey result JSON and then to continue the survey by importing the JSON to surveyJS. This works fine for all fields except the `file `question type.

### What is the expected behavior?
The user is able to see the attached file (show the filename only) again when he imports a JSON which contains the base64-encoded file.

An example of the question JSON and answer JSON is provided under the "Test code" section.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```
### Question JSON
      {
     type: "file",
     name: "file",
     title: "File Upload",
     storeDataAsText: false,
     maxSize: 10000000000
    },


**Answer JSON**
{
    "file": [
        "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABwYAAABuCAIAAACiMuCrAAAABmJLR0QAAAAAAAD5Q7t/AAAACXBIWXMAAA7EAAAOxAGVKw4bAAASzElEQVR4nO3dz27USNcH4NxTrmKua/ZzA7mCZDbvBphELFghpLBAiBUSI7F5F0hEn0bKOl9r0IuMXT4u/+nu8snzLFCPu1w+p7rTJf+mIRdPoaurq3jASp8+fdr1/ADQsjT7oEZas0kjf/75Z+9B/SkHv//++2+//Xao5K+//vrjjz/++eef9SUBAPBMXMRPi0QBYL/S7IMaaY1IFACAXROJAkBaafZBjbRGJAoAwK6JRAEgrTT7oEZaIxIFAGDXRKIAkFaafVAjrRGJAgCwayJRAEgrzT6okdaIRAEA2DWRKACklWYf1EhrRKIAAOyaSBQA0kqzD2qkNSJRAAB2TSQKAGml2Qc10hqRKAAAuyYSBYC00uyDGmmNSBQAgF0TiQJAWmn2QY20RiQKAMCuiUQBIK00+6BGWiMSBQBg10SiAJBWmn1QI60RiQIAsGsXn0JXV1fxAAAAWEAkCgDAufiWKACklWYf1EhrNmlEJAoAwLmIRAEgrTT7oEZaIxIFAGDXNohELy4ufv45PB47ZSR68atjXO7ntIvnP1JhlfPXX734ctev8HChjt348NJnn784sub0NYt2snVuXAvrcPGf/zvqyMPjmhMX/LTWW7nOvdqOV8kp3w+bfOjN2sgEcK3RSJdIFACAc1kbicZ3d5M3bCeORLtPHeMeeOW97li4fDIrI9E1F2ohotrEKSPRBdKscwJHjUQrT2n5/RZ8Smz+Nj5LJLrJDC38r8eT0UhrRKIAAOzaxpHo2Jd6xpwxEn0a3FgOv3RTf7B3vLgawbndehbXEBysaTn+z95s9ZnmcCnGFip+8/SuW9Nscdiw/njOWVMV+41P71VevPrYicGrWaxkbNGKwyp7j0cGL2hQz9irMHdtg6Z6FVaWEdRWWcYv4/9NLYff5fxxpPd9zwUjh6no8NUp1havVX2zlZerHNw9MutVK3Yxq87eRYunT5bRK/5p/nr2ahhOMkYA1xqNdIlEAQA4l1WR6PCOrng8mKGRSLRYcP3B4rDhKZPLEt/urqm2spj6Z4Peh8fr6wmWaPINNquLuSnD2FTFkYtXO17YBZ3GI4vDtpq85gUNeqwptb68yjVf8CM22U608p3Usvjtzu7B+pHDZyubGntqk7fEstULalt8iQV1Bm+nBT/scy9av3RjBHCt0UiXSBQAgHOZuJuaFYlOPh7adSQ6dIxIJbjKgtPjc1fW31Mz1diDyvWpb3byQsd4deYuwuSlZ82wZpXWTz75gsYdxcOWVVIcsH7O5aeEmebkweJUG0aiweCVb/JC/TNrW/MCbfV+mDV4wQfI8JSaF6tIANcajXSJRAEAOJeJG6pnHokOA77u8eHllk244HLFYXEZxWG9eeJSKxOH+IqVD+orj5uNe5xsubKLWSOD3icbD2aofL0qW6svvrKMsbUaLvjYS7C+vOKAWY2seYcUThn/Qmjvb76PRaLByGNHogveb2te2XjaoKr6c+e2NquM+sUJTuk1VVzJIgFcazTSJRIFAOBcRKL9p4Ib8rHTN5lwctHig1tlCpUF1CQpsy7UexDPX7k+NSu25tVZFu7Eq13TxZrXq35hZ6U5k2Vs/oIuKK84YNkrGz9b+8pOfSG0fuQZvyU6d/7Ji67/+Co6xkffVh8LlW8ekahGWiMSBQBg17aMRCuDgK5GfuP8+qhl5bljU20VUcUPFt+ixy0sK2xsqgUh17KwrFjAVtlHfZ0149e8JeKC109e/4LWf3RsWMmG3S0/Zf6/EFo8eKRviW77ftvk5+sEH87rPzRWfj6PtT/2AVtDANcajXTliEQv/zX21NiA4imXv9q81O51F1/iqOX1rhIPWDzV5UDNhbojj91+sYwTXLT+KmPv6prTfw5b1tTpF79Zp3lXbFXDsvdM/QdO5cjzvv2KRa75WN7wo3ITK5d3OFXwtukenxxZLKb3qd4TzDB5cPJFqZmzZirWm7jLSvYb57t6I4vH6w92n3oKb5vHrj52ZGVhwSsyrLZ3Qx5cIljA3oDi1Scv/TSwoNnisLHHY3NWTjU5/2QlwdXHTpw1f3GquOC5kxdHPo0sVLeYyRk2LG/sQWUji98hw7Pm/h754uOfk/SeLX/btPo9fIz325rV680261WLu6ipc8Hj4M0TrOdYncP/LB4ZI4BrjUa6EkSiNfdawX1acPAYN0KbzN/CTdqsAoaD60+vf+1yOM3KrFnAxIu/U0d9z6ycfOXIbc8dO/2oH/un/3nZ6orxsowdCYZVntIbXJxhbNoFH4Bbzckam0WiC559Om0kyl7U3+qzC15QWGnND1GafVAjrRGJPlV8/aQ77GlwbxPfGz+V7ouKA2oODo8HVcVN1Z+15mDQ+9jg4pzDUotHiuOD68ZXjJuaHLa4x2Jf9bMFizY2w+RSx+fGJQ2HVZY9t4V4ZPHZuMGg1GWLPFbP2BItrm3WQq1sZ7ievVOCOhePjMuIh42NKV63ssjimJrFDNZ/VsFxAfXFxC3E1x2bqv5IcNGxs4qXHhtWc5XiVEVxkTUD2NbaSPQp/ArY5LkiUXrEZ8l4QWGllT9EafZBjbRGJPr0773KU/Utzdwbtqc5N2BrbgjrT68ZuVXN3Qdrrt5tc2yd5y5I8dlZ/7l4QYpzjq1G5XW3WoRheXPPHWt5svLhyE3mv+wodvQ08/25bJFrqp07Z1xbzUJt0k7vQbySNUXOGrmm2jUrOVlk5UU3Kbg34XCS+rUKThmrtthU/J/xmMq1+lnV5OSTizbsvThnzfz1A9jWBpHoGiJRADieNPugRlojEv1p2T1V8cRZt3nxXdlkqfVzFmcYjiz2EjRYnCpeisk6Kxu5HJg1T2UXxRbm9j4cGTyYLCO+7rJ+668+PLjJOgc1rJw/rnNYQG9YceT6YjZcwGGRRz1l8uDYSs5dwOLI4cFNVm84/5oi16z/rHPj9Z9bTP3Mk9cN6hlbtAVrVX9k5SIUzSoynopNiEQBIK00+6BGWiMS/al4xzJ2z7PsXii4AevqDlhW1diccUljFx2rLZ6qZuRYv8GAYJ0ra6s/WLzKZJ2VI8ce9MbX97VyEYrPVi513EJ95WM11E9VWUllO7151i/yslch6GWsvFllTJ5SvPSsxoeXq5mnOHJ4cGW1vRM3KbJ+MYN24oNBm4uLqbz05ErG9Ywt2uRFg9kmj9S3Fk879uxkv/FUbEIkCgBppdkHNdIakehPk3dENY+LR+pvOMeqCuZfM2dX5UXnNlJ/yqz1n3vKsoPFq0xetPJg8KCofs2X9btmqeN66isfq6H+JaucP25w7NmaU+oPTi5g/VoNzSqj8pTKkVvNM9lsfOnFV9l2WSbPCtqZPBgs0bJi4pE181Q2G1yxZzimOHnQfuUi1KzqZPGLp2ITIlEASCvNPqiR1ohEf6q8Ray5laq5U4qnWjl//enLKllwcFadY6fHjaxZ5LFLBMu11SrFK1B/3ZWLUL/Uk+cuWPx4BTZZ5+KaL1ixNYtcv5KL51zZzvp6Jk+pL7Jy5Iart2Al45Fz139xwb0JlxXTezB3ebtPBctYs2g1LQTzxwUsW+rK4ms65UhEogCQVpp9UCOtEYn+FNzJ9AYM776KUwV3hsX7t5qDw2rj+8a5d48ryxseHJu8vuZip8UJaxakZsXi5TrGgsSTVM42VnPxivVLVHP1BZcIFnaTFsYWc3ik2GA8ybYVxitZ08tkO3EZY8XMmmes2uKclZNXjowXqv7xVkUuu1DluZUlLe56WMbYKcNne2OK3Y11HbdQvHQwfzAmmDY+WDQceTkQrwAbEokCQFpp9kGNtEYkOim+k3Gfk4aX8jnwKsNRzfoRW/zzuLufYp88JyASBYC00uyDGmmNSLRGHImeshKOofILQeyaVxmOasGP2F5+Koff/UzZ5t6JRAEgrTT7oEZaIxIFAGDXRKIAkFaafVAjrRGJAgCwayJRAEgrzT6okdaIRAEA2DWRKACklWYf1EhrRKIAAOyaSBQA0kqzD2qkNSJRAAB2TSQKAGml2Qc10hqRKAAAuyYSBYC00uyDGmmNSBQAgF0TiQJAWmn2QY20RiQKAMCuiUQBIK00+6BGWiMSBQBg10SiAJBWmn1QI60RiQIAsGsiUQBIK80+qJHWiEQBANg1kSgApJVmH9RIa0SiAADsmkgUANJKsw9qpDUiUQAAdk0kCgBppdkHNdIakegPX79+fffu3YsXLw5/Hh6fvgAAAJYRiQJAWmn2QY20RiR68OrVq7dv337+/Pn79++HPw+PX7582R1weXnZO2V4BACAsxCJAkBaafZBjbRGJPrly5c3b97891eHI4fj3WHdDFQeCgDQDpEoAKSVZh/USGtEoq9fv37//v3fv7q/vz8c7w4rRqKX/9MbOTwIAMCRiEQBIK00+6BGWiMSvbm5+bvk+vq6N/JHytnNQ3tPjR0EAOB4RKIAkFaafVAjrRGJ3t3dFb8lejg+HDyWeBYjUQAATkAkCgBppdkHNdIakejHjx/v7u56kejt7e3h+HBwLwbtGh4/RfUAAM+eSBQA0kqzD2qkNSLRx8fHQzG3t7f39/c/vh96eHxzc3M4Phw86+/FS0UBAE5AJAoAaaXZBzXSGpHowcPDw4cPH+7u7q6vrw9/Hh4fjhRHzvqL8yJRAIATEIkCQFpp9kGNtEYk+sPj4+PDw8O3b98Ofxa/H/pDzS+X9xfnAQBOSSQKAGml2Qc10hqRKAAAuyYSBYC00uyDGmmNSBQAgF27+BS6urqKBwAAwAIiUQAAzsW3RAEgrTT7oEZas0kjIlEAAM5FJAoAaaXZBzXSGpEoAAC79lwi0YuLfqfDIwCQjACuNRrpEokCAHAuzyUSffo1A5WHAvAcCOBao5EukSgAAOeyTSR68av6y7cQiRbLXtALALRGANcajXSJRAEAOJcNItFedDgrSTzxvyX6o7AgGw0OAsDuCOBao5EukSgAAOeyNhItBqD1qejpf73S8Nugw8diUAByEMC1RiNdCSLRr1+/vnv37sWLF4c/D49PXwAAAMtsEInOOt7TQiRa/Cv//uI8AAkI4Fqjka69R6KvXr16+/bt58+fv3//fvjz8Pjly5e9MZeXl5NHAAA4PZHoRJ1SUQD2SwDXGo107ToS/fLly5s3b/77q8ORw/HusMt/9Y6csk4AAIpEov4tUQDSEsC1RiNdu45EX79+/f79+79/dX9/fzjeHfYjAO3GoL3Hw8wUAIATeO7/luiT3zgPQF4CuNZopGvXkejNzc3fJdfX191hQSQ6lpMCAHACz+s3zgPAs5JmH9RIa0Sid3d3xW+JHo53hw0DUJEoAEALNohEn8Z/SdEkkSgAHE+afVAjrRGJfvz48e7urheJ3t7eHo53hw1zT5EoAEALtolEFxOJAsDxpNkHNdIakejj4+OhmNvb2/v7+x/fDz08vrm5ORzvDhOJAgC0SSQKAGml2Qc10hqR6MHDw8OHDx/u7u6ur68Pfx4eH470xgx/3bxIFACgBSJRAEgrzT6okdaIRH94fHx8eHj49u3b4c/e90N/GGadfuM8AEALLj6Frq6u4gEAALBAgkgUAICd8i1RAEgrzT6okdZs0ohIFACAcxGJAkBaafZBjbRGJAoAwK49o0j04qLf7PAIAGQigGuNRrpEogAAnMvzikR7GahIFIDcBHCt0UiXSBQAgHPZJhK9+FX95U//LdFueb3Hc4sHgMYJ4FqjkS6RKAAA57JBJNpLEmcFi41EomM5KQDsmgCuNRrpEokCAHAuayPRYgBan4qe5d8SDR48iUQBSEQA1xqNdIlEAQA4lw0i0VnHe87165V63xgViQKQkgCuNRrpEokCAHAuItHyUwCQgACuNRrpEokCAHAuzzQSffr1b/eLRAFISQDXGo10iUQBADiX5/hviRaP+I3zAOQjgGuNRrpEogAAnMsz+o3zAPDcpNkHNdIakSgAALu2QST61PmK5dwvWopEAeB40uyDGmmNSBQAgF3bJhJdTCQKAMeTZh/USGtEogAA7JpIFADSSrMPaqQ1IlEAAHZNJAoAaaXZBzXSGpEoAAC7dvEpdHV1FQ8AAIAFRKIAAJyLb4kCQFpp9kGNtGaTRkSiAACci0gUANJKsw9qpDUiUQAAdk0kCgBppdkHNdIakSgAALsmEgWAtNLsgxppjUgUAIBdE4kCQFpp9kGNtEYkCgDArolEASCtNPugRlojEgUAYNdEogCQVpp9UCOtEYkCALBrIlEASCvNPqiR1ohEAQDYNZEoAKSVZh/USGtEogAA7JpIFADSSrMPaqQ1IlEAAHZNJAoAaaXZBzXSGpEoAAC7JhIFgLTS7IMaaY1IFACAXROJAkBaafZBjbRGJAoAwK6JRAEgrTT7oEZaIxIFAGDXRKIAkFaafVAjrRGJAgCwayJRAEgrzT6okdaIRAEA2LX/B2FEn4WSf7VQAAAAAElFTkSuQmCC"
    ]
}

```

### Specify your

* browser:
* browser version:
* surveyjs platform (angular or react or jquery or knockout or vue): react
* surveyjs version: 1.0.11

**Comment by tsv2013** [comment]
@sfcrebecca Thank you for the suggestion
I'd like to clarify some details:
1) I think you meant
```JS
storeDataAsText: true,
```
setting. Am I right?
2) What place do you want the file name to be stored to? And Does it really make sence to store file names?

**Comment by sfcrebecca** [comment]
Thanks, @tsv2013 

1. Yes,  you are correct. Sorry for my mistake.

2. Actually the our design just wants to restore what the user input last time and then allows him/her to continue their editing by importing the saved JSON file. The current design creates a problem that everytime the user **re-edits** the form, they have to **re-upload** the file again in order to **keep** this file in their JSON file.

The rationale behind is that our form is quite long and the user may not be able to complete the form in one go. By having the save&reload functions, the users then are able to fill in the form multiple times and even pass the survey to another user for completing one section. 

For the file upload function, when we attach a file to the form, its filename is displayed next to the upload component. So in order to restore what the user input last time, we think storing the filename in a separate field (like the otherText?) is the only to achieve this.

If this creates some difficulties, it may also be acceptable to use the question name/title, such as  support_file01.xls, support_file02.pdf etc. Note that it is not required to view the file, just need to display the/a filename and save its content in the JSON after re-editing. 

Hope this clarifies our situation.

**Comment by tsv2013** [comment]
This issue is related to the https://github.com/surveyjs/surveyjs/issues/984 - Fileupload file names and storeDataAsText

**Comment by tsv2013** [comment]
Implemented in 1.0.19
